### ISSUE TITLE
Custom URLs in a multisite setup share the same URL across all nodes.

### ISSUE THREAD

**Issue by jlachancekffein** [verbb/navigation#405]
### Describe the bug

When the propagation method of a navigation is set to Save nodes to all sites enabled for this navigation, the modification of the url of a node of type Custom Url, the modification is applied to each site.

However, the label is correctly translated on each site. The issue is only with the URL.

I understand that, when I create a new node, that the same value is save in each others sites, but after modification, is suppose to apply only to the selected site.

For now, we fix it by changing the propagation mode to Only save nodes to the site they were created in but it break the relation between the navigation between site.

Im pretty sure that Craft, by default with the propagation mode set to Save nodes to all sites enabled, duplicate the content of the element in each site, but manage by sites after modification.

### Steps to reproduce

1. Create a custom URL on any site by filling in the title and URL.
2. Switch to another site in the Navigation plugin to edit the nodes.
3. Update the title and URL.
4. Return to the first site to edit the custom URL.
5. The title remains the same, but the URL is the one defined for the other site.

### Craft CMS version

5.4.2

### Plugin version

3.0.4

### Multi-site?

Yes

**Comment by engram-design** [comment]
This is a good point, and It's something that's been on my list to do for a while now. Essentially, we need to create a record of a node for each site enabled for a node.

While Craft does this automatically for a Node, because it's an element, there's only a few things that are stored site-specific through that. The `title`, `slug`, `uri` and `content` are site-specific for all Elements. You can see this in the `elements_sites` table.

However, the things that Navigation Nodes have such as the `url`, `classes`, `urlSuffix`, etc are not site-specific.

This'll be part of a larger refactor coming soon!

**Comment by MoritzLost** [comment]
@engram-design We've also run into this problem. Can you give an ETA for this large refactor?

I think Craft also solved this issue for a couple of native properties, like the `alt` field which was moved to the `elements_sites` table to be able to translate it.

**Comment by engram-design** [comment]
No ETA sorry, and yes those work due to them being stored as content due the field layout being localized.

**Comment by svonlanthen** [comment]
My Workaround:
When i have Sites like "English" and "French", and I want a Custom URL to be different per Site - I create two Custom URL Items and disable one of them after Switching the Site.
In this case i can keep all the other Navigation Items (mainly entries) in sync.

But took me a time to even identify the Problem - It's just not Userfriendly especially for clients.

But I can understand that you have to cope with given Limitations - No no offense and thank you for your work and contribution <3

**Comment by engram-design** [comment]
@svonlanthen Yep, appreciate it's a bit of a hassle, sorry. We were just waiting to see how Craft 6's plan was, and whether to wait for that. With Craft 5 slated for LTS, we'll probably roll this into the Craft 5 version as well.

**Comment by Megafry** [comment]
Any chance this will be added to the Craft 5 version?
My client is asking for this feature.

**Comment by engram-design** [comment]
Yes, it'll absolutely be added to the Craft 5 version. Timeline-wise, likely Q2/Q3 2026. Haven't forgotten about this, it's just more than a quick fix.

**Comment by nicolasbinet** [comment]
I just ran into this problem and this is a bummer.
I had to duplicate the whole navigation in order to make it work and deactivate some by version.
Hope this gets added to a soon to come version.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.1](https://github.com/verbb/navigation/releases/tag/4.0.0-beta.1). 

Be sure to check out the updated [docs](https://verbb.io/craft-plugins/navigation/docs/v4/get-started/installation-setup).
